src/test/java/org/javarosa/core/test/Scenario.java

```diff
@@ -502,4 +515,25 @@ public static AnswerResult from(int jrCode) {
                 .orElseThrow(RuntimeException::new);
         }
     }
+
+    public Scenario serializeAndDeserializeForm() throws IOException, DeserializationException {
+        // Initialize serialization
+        PrototypeManager.registerPrototypes(JavaRosaCoreModule.classNames);
+        PrototypeManager.registerPrototypes(CoreModelModule.classNames);
+        new XFormsModule().registerModule();
+
+        // Serialize form in a temp file
+        Path tempFile = createTempFile("javarosa", "test");
+        formDef.writeExternal(new DataOutputStream(newOutputStream(tempFile)));
+
+        // Create an empty FormDef and deserialize the form into it
+        FormDef deserializedFormDef = new FormDef();
+        deserializedFormDef.readExternal(
+            new DataInputStream(newInputStream(tempFile)),
+            PrototypeManager.getDefault()
+        );
+
+        delete(tempFile);
+        return new Scenario(deserializedFormDef, new FormEntryController(new FormEntryModel(deserializedFormDef)));
```

Comment:
Not really sure why the constructor takes in a FormEntryController since it's derived from a FormDef but it was already like that and I didn't change it.

Reply:
It's just because normally I wouldn't have any kind of computation in constructor methods and, since the controller is a final member, it has to be provided.

If creating an object from any set of arguments requires computation, I normally have a factory methods that deals with the computation and provides all the arguments for the single class contructor.

Reply:
> normally I wouldn't have any kind of computation in constructor methods

Ahh, interesting. That makes sense.

I think it's ok to leave like this for now.